source/common/stats/isolated_store_impl.cc

```diff
@@ -8,13 +8,13 @@
 #include "common/stats/fake_symbol_table_impl.h"
 #include "common/stats/histogram_impl.h"
 #include "common/stats/scope_prefixer.h"
+#include "common/stats/symbol_table_creator.h"
 #include "common/stats/utility.h"
 
 namespace Envoy {
 namespace Stats {
 
-IsolatedStoreImpl::IsolatedStoreImpl()
-    : IsolatedStoreImpl(std::make_unique<FakeSymbolTableImpl>()) {}
+IsolatedStoreImpl::IsolatedStoreImpl() : IsolatedStoreImpl(SymbolTableCreator::makeSymbolTable()) {}
```

Comment:
Use of SymbolTable for the IsolatedStoreImpl used by HostDescriptionImpl and ClusterInfoImpl::load_report_stats_store_ is likely to increase memory noticeably.

Can we continue using FakeSymbolTable for IsolatedStoreImpl?

Reply:
I assume the reason for this is that you've observed that there is no "." structure in the stat names used in those contexts? 

In that case I think you are right --- but there may be other contexts with isolated stats where it does make sense. Also right now isolated stores are used in unit tests for code that would otherwise use the main thread local stats store. Switching them to use fake symbol tables makes those tests behave less like prod 

I have an in-progress branch to try to add a stats memory test for large numbers of hosts. Maybe I can send you the diff a bit later and you can get it working?

Reply:
I assume the reason for this is that you've observed that there is no "." structure in the stat names used in those contexts? 

In that case I think you are right --- but there may be other contexts with isolated stats where it does make sense. Also right now isolated stores are used in unit tests for code that would otherwise use the main thread local stats store. Switching them to use fake symbol tables makes those tests behave less like prod 

I have an in-progress branch to try to add a stats memory test for large numbers of hosts. Maybe I can send you the diff a bit later and you can get it working?

Reply:
Also @antoniovicente it seems like the better solution is the one you proposed offline -- to make those particular stats just be some atomics and a reference/view to the string-literal name.